File: Palindromo Number/Palindromo Number.swift
//
//  Two_Sum.swift
//  Leetcode_parctice
//
//  Created by PixelShi on 2017/2/8.
//  Copyright © 2017年 sfmDev. All rights reserved.
//

import UIKit

class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        guard x >= 0 else {
            return false
        }

        var x = x
        var div = 1
        while (x/div >= 10) {
            div = div * 10
        }
        while x > 0 {
            print(div)
            let left = x / div
            print(left)
            let right = x % 10
            print(right)

            if left != right {
                return false
            }
            x = (x % div) / 10
            div = div / 100
        }
        return true
    }
}

File: Palindromo Number/Palindromo Number.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x<0:
            return False
        elif x==0:
            return True
        buff_x = x
        div = 1
        while x/div >= 10:
            div = div * 10

        while buff_x > 0:
            left = buff_x / div
            right = buff_x % 10
            if left != right:
                return  False
            buff_x = (buff_x % div) / 10
            div = div / 100
        return True


c = a.isPalindrome(0)
print(c)
